src/logic/graphMapping.ts

```diff
@@ -1,36 +1,121 @@
 import {
-  MappedEdge, 
+  MappedEdge,
   MappedNode,
   Edge,
   MappedGraph,
   GraphIndex,
   DisassembledGraph,
+  Position,
 } from './commonInterfaces';
+const rand = require('random-seed');
+
+const SWITCH_ALGORITHM_ABOVE_X_NODES = 10;
+const CIRCLE_RADIUS = 40;
 
 
 export default function mapGraph(graph: DisassembledGraph): MappedGraph {
   return calcNodes(graph[GraphIndex.NODES], graph[GraphIndex.EDGES]);
 }
 
 function calcNodes(nodes: Array<number>, edges: Array<Edge>):
-  [Array<MappedNode>, Array<MappedEdge>]
-{
-  // TODO better node placing
-  const mappedNodes = nodes.map((l, i) => ({
-    id: i,
-    x: Math.random() * 100,
-    y: Math.random() * 100,
-  }));
+  [Array<MappedNode>, Array<MappedEdge>] {
+
+  const mappedNodes = placeNodes(nodes, edges);
   const mappedEdges = edges.map((edge: Edge, i) => {
     const defaultNode = { x: 0, y: 0 };
-    const node1 = mappedNodes.find((e) => e.id === edge.from) ?? defaultNode;
-    const node2 = mappedNodes.find((e) => e.id === edge.to) ?? defaultNode;
+    const nodeFrom = mappedNodes.find((_, i) => i === edge.from) ?? defaultNode;
```

Comment:
What is `i`?

Reply:
iterator

Reply:
Then find a meaningful name for this

Reply:
If `edge.from` depends on some arbitrary iteration the something must be wrong

Reply:
No, list is ordered. `i` is meaningful name by convention.

Reply:
It actually stands for "index", internet says.

Reply:
Using `e.id` was way better then

Reply:
or rename `i` -> `id` if it is the same

Reply:
Noone uses `id` but will change